File: src/foundation/java/io/karma/pda/foundation/component/BlockImage.java
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

package io.karma.pda.foundation.component;

import io.karma.pda.api.app.component.AbstractComponent;
import io.karma.pda.api.app.component.ComponentType;
import io.karma.pda.api.state.MutableState;
import io.karma.pda.api.state.Synchronize;
import net.minecraft.world.level.block.Block;
import net.minecraft.world.level.block.Blocks;

import java.util.UUID;

/**
 * @author Alexander Hinze
 * @since 13/04/2024
 */
public class BlockImage extends AbstractComponent {
    @Synchronize
    public final MutableState<Block> block = MutableState.of(Blocks.DIRT);

    public BlockImage(final ComponentType<?> type, final UUID id) {
        super(type, id);
    }
}

File: src/composable/kotlin/io/karma/pda/composition/component/BlockImage.kt
/*
 * Copyright (C) 2024 Karma Krafts & associates
 */

@file:JvmName("ComponentExtensionsKt")
@file:JvmMultifileClass

package io.karma.pda.composition.component

import io.karma.pda.api.app.component.Container
import io.karma.pda.composition.Composable
import io.karma.pda.composition.Composer
import io.karma.pda.composition.LayoutProps
import io.karma.pda.foundation.component.BlockImage
import io.karma.pda.foundation.component.DefaultComponents

/**
 * @author Alexander Hinze
 * @since 07/06/2024
 */

@Composable
inline fun <C : Container> Composer<C>.BlockImage(
    crossinline layoutProps: LayoutProps,
    localName: String? = null,
    props: @Composable BlockImage.() -> Unit = {}
) = component(
    type = DefaultComponents.BLOCK_IMAGE,
    layoutProps = layoutProps,
    localName = localName,
    props = props
)